app/uk/gov/hmrc/exports/models/declaration/submissions/Action.scala

```diff
@@ -18,17 +18,69 @@ package uk.gov.hmrc.exports.models.declaration.submissions
 
 import play.api.libs.functional.syntax._
 import play.api.libs.json._
+import uk.gov.hmrc.exports.models.declaration.ExportsDeclaration
+import uk.gov.hmrc.exports.models.declaration.submissions.RequestType.RequestTypeFormat
 
 import java.time.{LocalDateTime, ZoneId, ZonedDateTime}
 
-case class Action(
-  id: String,
-  requestType: RequestType,
-  requestTimestamp: ZonedDateTime = ZonedDateTime.now(ZoneId.of("UTC")),
-  notifications: Option[Seq[NotificationSummary]] = None
-) {
+sealed trait Action {
+
+  val id: String
+  val notifications: Option[Seq[NotificationSummary]]
```

Comment:
Is there a reason that `decId` is not an optional string declared in the trait?

Reply:
> Where does the `submission.latestVersionNo` get updated by the way, can't seem to find that or a test for it?
> 
> Looking pretty good though!

There's no context for it at the moment. As we only iterate on the version for amendments, so far we're only dealing with submission and cancellation so there's no need for it

Reply:
> 

The `decId` isn't a property of all actions. 

The `decId` is a mandatory String in all actions that need it, and it does not exist in the actions that do not.

Rather than being an optional String because it either exists or it doesn't based on the request type